rfcs/0018-glides-trainsheet-events.md

```diff
@@ -0,0 +1,538 @@
+- `endLocation` (Location, conditionally required): where the trip will be ending. Required if `tripType` is `"arrival"`.
+- `time` (Time, conditionally required): when the added trip is expected to depart (`tripType` of `"departure"`) or arrive (`tripType` of `"arrival"`). Required when `tripType` is `"departure"` or if `previousTripKey` is not specified.
+- `tripType` (TripType): whether `time` for the new trip is the when the trip is departing `startLocation` (`"departure"`) or when the trip is arriving at `endLocation` (`"arrival"`).
+- `previousTripKey` (Trip Key, optional): links the newly created trip to the trip immediately before it. If the Trip Key refers to an added trip, it may not have been seen in the event stream at the time of this message.
+- `nextTripKey` (Trip Key, optional): links the newly created trip to the trip after it. If the Trip Key refers to an added trip, it may not have been seen in the event stream at the time of this message
+- `consist` (array of Optional Car, optional, default: empty list): the cars assigned to perform this trip. If a car is `null`, then no car is currently assigned to that position in the consist.
+- `operators` (list of Optional Operator, optional, default: empty list): the array of operators assigned to perform this trip. If an operator is `null`, then no operator is assigned to the car in that position in the consist.
+- `comment` (string, optional): free text with additional information about the trip
+
+*Notes*
+- The event should only include values which were explicitly included by the `author`: inferred values should not be included.
+- `consist` and `operators` MUST be the same length; if they are not, the behavior of consumers is undefined.
+
+### Trip arrived
+Inspectors are required to note when trains arrive at their location. This event is generated by that confirmation, and includes the relevant information about who performed the trip. 
+
+Event type: `com.mbta.ctd.glides.trip-arrived`
+Fields in the event:
+- `author` (Author): the inspector adding the trip.
+- `tripKey` (Trip Key): which trip is being confirmed.
+- `comment` (string, optional): free text information about the trip.
+- `location` (Location): the location at which the train has arrived.
+- `arrivalTime` (Time): if present, the time that the train was recorded arriving at the specified location.
+- `consist` (array of Car): the cars which performed this trip.
+- `operators` (array of Operator): the list of operators who to performed this trip.Even if the scheduled operators performed the trip, those values MUST be included here.
+
+*Notes*
+- `consist` and `operators` MUST be the same length; if they are not, the behavior of consumers is undefined.
+
+### Trip departed
+Inspectors are required to note when trains arrive at their location. This event is generated by that confirmation, and includes the relevant information about who performed the trip. 
```

Comment:
When should Glides publish the `departed` intent rather than `updated`? Sometimes inspectors will set a planned time minutes in advance, before figuring out other details like the consist and operators, and then later make an update without changing the time. In that case is it okay to publish `departed` even though the consist and operators aren't final, and then an `updated` later? Do we publish an `updated` if the time is set ahead of time and then wait until the time comes around to re-publish a `departed` reacting to a time instead of an interaction? If an inspector re-edits the departure time, do we publish a second `departed` or an `updated`?

The idea of capturing an inspector's intent is good, and if we have to do it, Glides should be the app do it. But we've been building the trainsheets as if their purpose is knowing information rather than reacting to events/intent, so I don't think we can really handle those questions well.